yt_geotiff/data_structures.py

```diff
@@ -539,10 +603,44 @@ def _is_valid(self, *args, **kwargs):
 
 class JPEG2000Dataset(GeoTiffDataset):
     _index_class = JPEG2000Hierarchy
-    _valid_extensions = ('.jp2')
+    _field_info_class = JPEG2000FieldInfo
+    #_valid_extensions = ('.jp2',)
     _driver_type = "JP2OpenJPEG"
     _dataset_type = "JPEG2000"
 
+class RasterioGroupDataset(GeoTiffDataset):
+    _dataset_type = "RasterioGroup"
+    _valid_extensions = ('.tif','.tiff','.jp2')
+    _index_class = RasterioGroupHierarchy
+    _field_info_class = JPEG2000FieldInfo
+
+
+    # list of all filenames in directory
+    
+    @classmethod
+    def _is_valid(self, *args, **kwargs):
+        # for number of files
+        for fn in args:    
+        
+            # if not instance fnlist
+            #    return false
+            
+            valid = False
+            for ext in self._valid_extensions:
+                if re.search(f"{ext}$", fn, flags=re.IGNORECASE): 
+                    valid = True
+                    break
+            if not valid:
+                return False        
+        return True
+
+    def __init__(self, *args, field_map=None):
+        self.filename_list = args
+        filename = args[0] 
+        super().__init__(filename, field_map=field_map)
+        self.data = self.index.grids[0]
```

Comment:
This line happens in the superclass's init function as well, so it probably is not necessary here.

Reply:
It seems this line has to be present otherwise the following error is returned:
AttributeError: 'RasterioGroupDataset' object has no attribute 'dataset_type'